### PR TITLE
49 document api grid

### LINKED ISSUE #49: Document API: grid
Document all of the files in the functions in the src/grid directory

### PR DESCRIPTION
**Pull request by K20shores**
Link to [this branch's docs with my changes](https://ncar.github.io/photo-decomp/branch/49-document-api-grid/api/grid/).

This should close #49. Notably missing in this PR is the description of the configuration for grids, but that is because #48 seems to be a catch-all for configuration. I will certainly add the configuration information if it is desired here, but I thought this was specifically for the code documentation.

### CODE REVIEW

File: docs/source/api/grid.rst [hunk 1]
```diff
@@ -0,0 +1,43 @@
+.. TUV-x Grids
+
+Grids
+=====
+
+A :f:type:`~tuvx_grid/grid_t` object defines the vertical extent of the model.
```
Comment (mattldawson) [inline docs/source/api/grid.rst:6]:
```suggestion
A :f:type:`~tuvx_grid/grid_t` object defines discreet values along an axis (e.g., height, wavelength). 
```

File: docs/source/api/grid.rst [hunk 2]
```diff
@@ -0,0 +1,43 @@
+.. TUV-x Grids
+
+Grids
+=====
+
+A :f:type:`~tuvx_grid/grid_t` object defines the vertical extent of the model. 
+There are three 
+classes that can be used to configure TUV-x. Typically, the type of grid used
+at runtime is configurd from the configuration file. See
```
Comment (mattldawson) [inline docs/source/api/grid.rst:9]:
```suggestion
at runtime is configured from the configuration file. See 
```

File: docs/source/api/grid.rst [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+There are three 
+classes that can be used to configure TUV-x. Typically, the type of grid used
+at runtime is configurd from the configuration file. See 
+:ref:`configuration-grid` for more information.
+
+All grid configurations are generated by the factory pattern inside the
+:f:mod:`tuvx_grid_factory`. Each grid is 
+created once and lives in the :f:type:`~tuvx_grid_warehouse/grid_warehouse_t` 
+class. A copy of the grid is returned each time the grid is needed. Each class
+the depends on a grid will take a
```
Comment (mattldawson) [inline docs/source/api/grid.rst:16]:
```suggestion
that depends on a grid will take a 
```

File: docs/source/api/grid.rst [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+      integer                               :: thing
+      type(grid_warehouse_t), intent(inout) :: grid_warehouse
+      class(grid_t),          pointer       :: zGrid => null( )
+
+      zGrid => grid_warehouse%get_grid( "height", "km" )
+      ...
+    end function run
+
+.. f:automodule:: tuvx_grid
```
Comment (mattldawson) [inline docs/source/api/grid.rst:33]: Not sure if this is fixable, but it seems like the formatting of the module description works where the module is described:
image

but when the module is used in another module/function, the formatting symbols end up in the text:
image

Comment (K20shores) [reply docs/source/api/grid.rst:33]: Ya, I noticed that. I couldn't find a way to fix this. The alternative is to remove the headings from the module and only have the description there. I can do that and then maybe these short descriptions would be more useful.

Comment (K20shores) [reply docs/source/api/grid.rst:35]: Done.

**Comment by K20shores** [comment]
@mattldawson this should be good to go now

File: src/grid_warehouse.F90 [hunk 5]
```diff
@@ -121,10 +113,10 @@ function get_grid_string( this, name, units ) result( grid_ptr )
...
     use tuvx_grid,                     only : grid_t
 
-    class(grid_warehouse_t), intent(inout) :: this
-    type(string_t),          intent(in)    :: name
-    type(string_t),          intent(in)    :: units
-    class(grid_t), pointer                 :: grid_ptr
+    class(grid_warehouse_t), intent(inout) :: this ! This :f:type:`~tuvx_grid_warehouse/grid_warehouse_t`
+    type(string_t),          intent(in)    :: name ! The name of a grid, see :ref:`configuration-grid` for grid names
+    type(string_t),          intent(in)    :: units ! The units of the grid 
+    class(grid_t), pointer                 :: grid_p ! The :f:type:`~tuvx_grid/grid_t` which matches the name passed intr
```
Comment (mattldawson) [inline src/grid_warehouse.F90:119]:
```suggestion
    class(grid_t), pointer                 :: grid_ptr ! The :f:type:`~tuvx_grid/grid_t` which matches the name passed intr  
```

**Review by mattldawson — APPROVED** [review]
looks good to me!

**Pull request merged**